spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/route/RouteDefinition.java

```diff
@@ -115,6 +119,14 @@ public void setOrder(int order) {
 		this.order = order;
 	}
 
+	public Map<String, Object> getMetadata() {
+		return metadata;
+	}
+
+	public void setMetadata(Map<String, Object> metadata) {
```

Comment:
I think this is miss named. `setMetadata()` should do `this.metadata = metadata`. If user wants to put all, call `getMetadata().putAll()`.

Reply:
The API which force to use getMetadata().putAll(), it is not the API which I'd expect to see
Maybe It will be better to add method 
```
public void clearMetadata(){
this.metadata = new HashMap<>();
}
```
From the user, perspective metadata is going to be used to contain all metadata from all places and need to combine them but not replace by default.

Or I may rework setMetadata as you ask and add separate  methods
```
public void addMetadata(Map<String, Object> metadata) {
		this.metadata.putAll(metadata);
	}
public void addMetadata(String key, Object value) {
		this.metadata.put(key, value);
	}
```
What do you think?

Reply:
I've changed setMetadata as you asked
and added 2 addMetadata methods.
Hope you don't mind.